niworkflows/workflows/epi/refmap.py

```diff
@@ -0,0 +1,220 @@
+    fit an exponential decay to model the signal drop caused by the increasing
+    temperature of the device (this is heavily related to BOLD *nonsteady states*
+    described above, as these are just the few initial instants when the exponential
+    decay is much faster).
+
+    Workflow Graph
+        .. workflow::
+            :graph2use: orig
+            :simple_form: yes
+
+            from niworkflows.workflows.epi.refmap import init_epi_reference_wf
+            wf = init_epi_reference_wf(omp_nthreads=1)
+
+    Parameters
+    ----------
+    omp_nthreads : :obj:`int`
+        Maximum number of threads an individual process may use
+    name : :obj:`str`
+        Name of workflow (default: ``epi_reference_wf``)
+    auto_bold_nss : :obj:`bool`
+        If ``True``, determines nonsteady states in the beginning of the timeseries
+        and selects them for the averaging of each run.
+        IMPORTANT: this option applies only to BOLD EPIs.
+
+    Inputs
+    ------
+    in_files : :obj:`list` of :obj:`str`
+        List of paths of the input EPI images from which reference volumes will be
+        selected, aligned and averaged.
+
+    Outputs
+    -------
+    epiref : :obj:`str`
+        Path of the generated EPI reference file.
+    xfms : :obj:`list` of :obj:`str`
+        List of rigid-body transforms in LTA format to resample from
+        the reference volume of each run into the ``epiref`` reference.
+    volumes : :obj:`list` of :obj:`str`
+        List of paths to the reference volume generated per input run.
+    gs_drift : :obj:`list` of :obj:`list` of :obj:`float`
+        A list of global signal drift factors for the set of volumes selected
+        for averaging, per run.
+    n_dummy : :obj:`list` of :obj:`int`
+        Number of nonsteady states at the beginning of each run (only BOLD with
+        ``auto_bold_nss=True``)
+
+    See Also
+    --------
+    Discussion and original flowchart at `nipreps/niworkflows#601
+    <https://github.com/nipreps/niworkflows/issues/601>`__.
+
+    """
+    from nipype.interfaces.ants import N4BiasFieldCorrection
+    from ...utils.connections import listify
+
+    from ...interfaces.bold import NonsteadyStatesDetector
+    from ...interfaces.freesurfer import StructuralReference
+    from ...interfaces.header import ValidateImage
+    from ...interfaces.images import RobustAverage
+    from ...interfaces.nibabel import IntensityClip
+
+    wf = Workflow(name=name)
+
+    inputnode = pe.Node(
+        niu.IdentityInterface(fields=["in_files", "t_masks"]), name="inputnode"
+    )
+    outputnode = pe.Node(
+        niu.IdentityInterface(
+            fields=["epiref", "xfms", "volumes", "gs_drift"]
+            + ["n_dummy"] * auto_bold_nss
+        ),
+        name="outputnode",
+    )
+
+    validate_nii = pe.MapNode(
+        ValidateImage(), name="validate_nii", iterfield=["in_file"]
+    )
+
+    run_avgs = pe.MapNode(
```

Comment:
`run_avgs` feels like "run the averages", not "averages of the runs". What about?

```suggestion
    per_run_avgs = pe.MapNode(
```